### PR TITLE
fix duplicate project references fail GetCompilationAsync

### PR DESCRIPTION
**Pull request by superyyrrzz**
#1414

### CODE REVIEW

File: src/Microsoft.DocAsCode.Metadata.ManagedReference/ExtractMetadataWorker.cs [hunk 1]
```diff
@@ -342,6 +342,11 @@ private static ExtractMetadataInputModel ValidateInput(ExtractMetadataInputModel
                     .Where(s => s is PortableExecutableReference)
                     .Select(s => ((PortableExecutableReference)s).FilePath));
                 FillProjectDependencyGraph(projectCache, projectDependencyGraph, project);
+                // duplicate project references will fail Project.GetCompilationAsync
+                if (project.ProjectReferences.GroupBy(r => r.ProjectId).Any(g => g.Count() > 1))
+                {
+                    projectCache[path] = project.WithProjectReferences(project.ProjectReferences.Distinct());
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.Metadata.ManagedReference/ExtractMetadataWorker.cs]: is it true that the equality of ProjectReferences.ProjectId means the equality of ProjectReference?

**Comment by vwxyzh** [comment]
Also add unit test?

File: src/Microsoft.DocAsCode.Metadata.ManagedReference/ExtractMetadataWorker.cs (hunk 1 above)
Comment (superyyrrzz) [inline src/Microsoft.DocAsCode.Metadata.ManagedReference/ExtractMetadataWorker.cs]: Yes, Roslyn generates it and keeps it unique.

File: test/docfx.Tests/MetadataCommandTest.cs [hunk 2]
```diff
@@ -256,5 +256,71 @@ public void TestMetadataCommandFromCSProjectWithFilterInOption()
             Assert.Equal(1, memberViewModel.Items.Count);
             Assert.NotNull(memberViewModel.References.Find(s => s.Uid.Equals("Foo")));
         }
+
+
+        [Fact]
```
Comment (hellosnow) [inline test/docfx.Tests/MetadataCommandTest.cs]: >Fact [](start = 9, length = 4)

minor: remove extra line

**Comment by hellosnow** [comment]
:shipit:

**Comment by vicancy** [comment]
:shipit:

**Pull request merged**